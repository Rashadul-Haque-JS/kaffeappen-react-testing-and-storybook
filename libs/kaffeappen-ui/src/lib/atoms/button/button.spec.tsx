import { render } from '@testing-library/react';
import ButtonComponent from './button';


describe('Button', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<ButtonComponent variant={'primary'} marginTop={40}>'Logga in'</ButtonComponent>);
    expect(baseElement).toBeTruthy();
  });

  it('should render successfully', () => {
    const { baseElement } = render(<ButtonComponent variant={'secondary'} marginTop={2}>'Registrera'</ButtonComponent>);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot successfully', () => {
    const { baseElement } = render(<ButtonComponent variant={'primary'} marginTop={40}>'Logga in'</ButtonComponent>);
    expect(baseElement).toMatchSnapshot();
  });

  it('should match snapshot successfully', () => {
    const { baseElement } = render(<ButtonComponent variant={'secondary'} marginTop={2}>'Registrera'</ButtonComponent>);
    expect(baseElement).toMatchSnapshot();
  });
});