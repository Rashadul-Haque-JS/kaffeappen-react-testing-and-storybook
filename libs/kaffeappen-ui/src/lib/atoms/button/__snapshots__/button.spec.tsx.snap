// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot successfully 1`] = `
<body>
  <div>
    <button
      class="button__StyledButton-sc-1l6wl6y-0 jakkFT"
    >
      'Logga in'
    </button>
  </div>
</body>
`;

exports[`Button should match snapshot successfully 2`] = `
<body>
  <div>
    <button
      class="button__StyledButton-sc-1l6wl6y-0 gBIovO"
    >
      'Registrera'
    </button>
  </div>
</body>
`;
